include/slang/ast/expressions/AssertionExpr.h

```diff
@@ -91,12 +96,22 @@ class SLANG_EXPORT AssertionExpr {
     /// and false otherwise.
     bool admitsEmpty() const;
 
+    bool isAdmitsOnlyEmpty() const { return admitsOnlyEmpty; }
+
+    /// Returns true if this is a sequence expression that admits no match,
+    /// and false otherwise.
+    bool admitsNoMatch() const;
+
+    /// Compute possible clock ticks length of sequence under assertion expression
+    std::optional<SequenceRange> computeSequenceLength() const;
+
     static const AssertionExpr& bind(const syntax::SequenceExprSyntax& syntax,
                                      const ASTContext& context, bool allowDisable = false);
 
     static const AssertionExpr& bind(const syntax::PropertyExprSyntax& syntax,
                                      const ASTContext& context, bool allowDisable = false,
-                                     bool allowSeqAdmitEmpty = false);
+                                     bool isFollowedByProp = false, bool isOverlapImpl = false,
```

Comment:
This is becoming too many boolean arguments in a row, which makes it easy to mess up the ordering of them when calling the function. I think this should be made a separate flags enum.

Reply:
fixed